libs/execution_context/ebpf_maps.c

```diff
@@ -2943,5 +2945,16 @@ ebpf_map_get_next_key_and_value_batch(
 
     *key_and_value_length = output_length;
 
+    if ((flags & EBPF_MAP_FIND_FLAG_DELETE) && (previous_key != NULL) && (output_length != 0)) {
+        // If the caller requested deletion, delete the last entry.
+        ebpf_result_t delete_result =
+            ebpf_map_metadata_tables[map->ebpf_map_definition.type].delete_entry(map, previous_key);
+        if (delete_result != EBPF_SUCCESS) {
+            EBPF_LOG_MESSAGE_UINT64(
```

Comment:
same comment as above.

Reply:
> same comment as above.

It is better to have an explicit error message "Failed to delete last entry", rather than generic error 'returned error'.